python/docs/examples.rst

```diff
@@ -1,17 +1,18 @@
 Examples
 ========
 
-Pandas
-*******
+Using Pandas DataFrames as Input / Output
+*******************************************
 
 
 Working wth pandas DataFrames in Feldera is fairly straight forward. 
-You can use :meth:`.SQLContext.connect_source_pandas` to connect a 
+You can use :meth:`.SQLContext.input_pandas` to connect a
 DataFrame to a feldera table as the data source. 
 
-To listen for response from feldera, in the form of DataFrames, it is necessary
-to to call :meth:`.SQLContext.listen` before you call 
-:meth:`.SQLContext.start` or :meth:`.SQLContext.run_to_completion`.
+To listen for response from feldera, in the form of DataFrames
+call :meth:`.SQLContext.listen`.
+To ensure all data is received start listening before calling
+:meth:`.SQLContext.start`.
```

Comment:
This could also be enforced via a state check in the `listen()` command?

Reply:
You can also start listening at some arbitrary point after starting a pipeline. 

Or listen to an already running pipeline.

Reply:
True, that's possible, my question is more whether a Python user with `SQLContext` ever wants specifically that, as there are no guarantees at what point of the stream listening will start. Probably out of scope for this PR.